src/libraries/System.Net.Security/src/System/Net/Security/TlsCipherSuiteNameParser.ttinclude

```diff
@@ -150,6 +150,28 @@ internal static class EnumHelpers
                 throw new Exception($"Value `HashAlgorithmType.{val}` does not have framework corresponding name. See TlsCipherSuiteNameParser.ttinclude: ToFrameworkName.");
         }
     }
+
+    public static int GetHashSize(HashAlgorithmType hash)
+    {
+        switch (hash)
+        {
+            case HashAlgorithmType.Aead:
+            case HashAlgorithmType.None:
+                return 0;
+            case HashAlgorithmType.Md5:
+                return 128;
+            case HashAlgorithmType.Sha1:
+                return 160;
+            case HashAlgorithmType.Sha256:
+                return 256;
+            case HashAlgorithmType.Sha384:
+                return 384;
+            case HashAlgorithmType.Sha512:
+                return 512;
+            default:
+                throw new ArgumentOutOfRangeException(Enum.GetName(hash));
```

Comment:
ArgumentOutOfRangeException takes `paramName` as the argument. This is passing argument value.

Also, `hash.ToString` would be better here. `Enum.GetName<TEnum>` is aggressively inlined generic method that expands into a ton of code. We do not need that here - this is exceptional path.

Reply:
Ah, yes I added it for testing (because the missing `Aead` case was missing). BTW, this is the generator code, it doesn't get baked into the final .cs file. The effect is only when t4.exe is running. :)